### PR TITLE
in embindgen.py added inpaint function

### PR DESCRIPTION
**Pull request by ilbalance**
In embindgen.py "photo" category added "inpaint" function.

```
force_builders=Custom
buildworker:Custom=linux-1,linux-2,linux-4
docker_image:Docs=docs-js
docker_image:Custom=javascript
```

### CODE REVIEW

**Comment by alalek** [comment]
This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

**Comment by dkurt** [comment]
@alalek, The thing is that `photo` module is added only in the master branch.

File: platforms/js/build_js.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def get_cmake_cmd(self):
         return cmd
 
     def get_build_flags(self):
-        flags = ""
+        flags = "-s USE_PTHREADS=0 "
```
Comment (dkurt) [inline platforms/js/build_js.py]: Please keep this unchanged

File: modules/js/src/embindgen.py [hunk 2]
```diff
@@ -98,7 +98,7 @@
              'compare', 'convertScaleAbs', 'copyMakeBorder', 'countNonZero', 'determinant', 'dft', 'divide', 'eigen', \
              'exp', 'flip', 'getOptimalDFTSize','gemm', 'hconcat', 'inRange', 'invert', 'kmeans', 'log', 'magnitude', \
              'max', 'mean', 'meanStdDev', 'merge', 'min', 'minMaxLoc', 'mixChannels', 'multiply', 'norm', 'normalize', \
-             'perspectiveTransform', 'polarToCart', 'pow', 'randn', 'randu', 'reduce', 'repeat', 'setIdentity', 'setRNGSeed', \
+             'perspectiveTransform', 'polarToCart', 'pow', 'randn', 'randu', 'reduce', 'repeat', 'rotate', 'setIdentity', 'setRNGSeed', \
```
Comment (dkurt) [inline modules/js/src/embindgen.py]: Please remove rotate as well. It will be merged separately with test

**Pull request merged**